src/python_testing/TC_OCC_2_1.py

```diff
@@ -0,0 +1,216 @@
+        else:
+            logging.info("Test step skipped")
+
+        self.step(4)
+        if attributes.OccupancySensorType.attribute_id in attribute_list:
+            occupancy_sensor_type_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorType)
+
+            if occupancy_sensor_type_dut != NullValue:
+                asserts.assert_less(occupancy_sensor_type_dut, Clusters.Objects.OccupancySensing.Enums.OccupancySensorTypeEnum.kUnknownEnumValue,
+                                    "OccupancySensorType is not in valid range")
+        else:
+            logging.info("Test step skipped")
+
+        self.step(5)
+        if attributes.OccupancySensorTypeBitmap.attribute_id in attribute_list:
+            occupancy_sensor_type_bitmap_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.OccupancySensorTypeBitmap)
+
+            if occupancy_sensor_type_bitmap_dut != NullValue:
+                asserts.assert_less_equal(occupancy_sensor_type_bitmap_dut, 0b00000111, "OccupancySensorTypeBitmap attribute is not in valid range")
+        else:
+            logging.info("Test step skipped")
+
+        self.step(6)
+        if attributes.HoldTime.attribute_id in attribute_list:
+            hold_time_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTime)
+
+            if hold_time_dut != NullValue:
+                asserts.assert_less_equal(hold_time_dut_dut, 0xFFFE, "HoldTime attribute is out of range")
+                asserts.assert_greater_equal(hold_time_dut_dut, 0, "HoldTime attribute is out of range")
+        else:
+            logging.info("Test step skipped")
+
+        self.step(7)
+        if attributes.HoldTimeLimits.attribute_id in attribute_list:
+            hold_time_limits_dut = await self.read_occ_attribute_expect_success(endpoint=endpoint, attribute=attributes.HoldTimeLimits)
+
+            if hold_time_limits_dut != NullValue:
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeMin, hold_time_limits_dut.HoldTimeMax, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeMin, 0, "HoldTimeMin is not in valid range")
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeMax, 0xFFFE, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeMax, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+                asserts.assert_less_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMax, "HoldTimeMin is not in valid range")
+                asserts.assert_greater_equal(hold_time_limits_dut.HoldTimeDefault, hold_time_limits_dut.HoldTimeMin, "HoldTimeMin is not in valid range")
+        else:
+            logging.info("Test step skipped")

```

Comment:
if you change these logging calls to self.mark_current_step_skipped(), it'll log and also notify the TH that the step was skipped.

Reply:
Does self.mark_current_step_skipped() support a string display? I want to add some string information that could be unique to the test step, so I will sometimes resort to using logging.info().

Reply:
It doesn't internally, because that's just for the TH GUI. You can still log, just call skipped too. Here, you're not adding any additional information though.

Reply:
ok with a follow up here, btw. If you want to do that, just drop an issue in here and close the comment.

Reply:
Will triage and resolve in the new issue filed at https://github.com/project-chip/connectedhomeip/issues/34686